Code/Framework/AzFramework/Tests/DocumentPropertyEditor/AdapterBuilderTests.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Tests/DocumentPropertyEditor/DocumentPropertyEditorFixture.h>
+#include <AzFramework/DocumentPropertyEditor/AdapterBuilder.h>
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h>
+#include <AzCore/DOM/DomUtils.h>
+#include <AzCore/std/any.h>
+
+namespace AZ::DocumentPropertyEditor::Tests
+{
+    using AdapterBuilderTests = DocumentPropertyEditorTestFixture;
+
+    TEST_F(AdapterBuilderTests, VisitSimpleStructure)
+    {
+        AdapterBuilder builder;
+        builder.BeginAdapter();
+        builder.BeginRow();
+        builder.Label("label");
+        builder.BeginPropertyEditor("TextEditor", Dom::Value("lorem ipsum", true));
+        builder.Attribute(AZ::Name("attr"), Dom::Value(2));
+        builder.EndPropertyEditor();
+        builder.EndRow();
+        builder.EndAdapter();
+
+        Dom::Value result = builder.FinishAndTakeResult();
+
+        /**
+        Expect the following structure:
+        <Adapter>
+            <Row>
+                <Label>label</Label>
+                <PropertyEditor type="TextEditor" attr=2>value</TextEditor>
+            </Row>
+        </Adapter>
+        */
+        Dom::Value expectedValue = Dom::Value::CreateNode("Adapter");
+        Dom::Value row = Dom::Value::CreateNode("Row");
+        Dom::Value label = Dom::Value::CreateNode("Label");
+        label.SetNodeValue(Dom::Value("label", true));
```

Comment:
repeated: This feels like it should be done in 1 line. If a way to construct a Dom::Value by providing the text and type is not available yet, it might be good to add it as I can see this as a pretty common operation.

Reply:
Hmm, I'll keep an eye out and see how comment it is to create an XML-style node and want to specify its content at the same time (not necessarily text, just some `Dom::Value`), it's tidy-up we can do later. In Javascript, I don't believe `document.createElement` lets you do this